tests/src/globalSetup/globalSetup.ts

```diff
@@ -0,0 +1,42 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { removeFolderIfExists } from '../utility/testUtility';
+
+let setupCalled = false;
+let teardownCalled = false;
+
+export async function setup() {
+  if (!setupCalled) {
+    console.log('Entering global setup');
+    // remove all previous testing output files
+    await removeFolderIfExists('tests/output');
+  } else {
+    console.log('Global setup already called, skipping...');
+  }
+  setupCalled = true;
+}
+
+export async function teardown() {
+  if (!teardownCalled) {
+    console.log('Entering global teardown');
+  } else {
+    console.log('Global teardown already called, skipping...');
+  }
+  teardownCalled = true;
```

Comment:
I think the logic is odd

it always updated teardownCalled to true

Reply:
absolutely, more for testing purposes.